tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
+            # section 2
+            qml.RY(params[1], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            # section 3
+            qml.RX(params[2], wires=0)
+            qml.RY(params[3], wires=1)
+            qml.RZ(params[4], wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            # section 4
+            qml.RX(params[5], wires=0)
+            qml.RY(params[6], wires=1)
+            qml.RZ(params[7], wires=2)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            return qml.expval(qml.PauliX(0))
+
+        circuit = qml.QNode(circuit, dev)
+
+        params = np.ones([8])
+        circuit.construct_metric_tensor([params])
+        res = circuit._metric_tensor_subcircuits
+
+        # this circuit should split into 4 independent
+        # sections or layers when constructing subcircuits
+        assert len(res) == 4
+
+        # first layer subcircuit
+        assert len(res[(0,)]['queue']) == 0
+        assert isinstance(res[(0,)]['observable'][0], qml.PauliX)
+
+        # second layer subcircuit
+        assert len(res[(1,)]['queue']) == 1
+        assert isinstance(res[(1,)]['queue'][0], qml.RX)
+        assert isinstance(res[(1,)]['observable'][0], qml.PauliY)
+
+        # third layer subcircuit
+        layer = res[(2, 3, 4)]
+        assert len(layer['queue']) == 4
+        assert len(layer['observable']) == 3
+        assert isinstance(layer['queue'][0], qml.RX)
+        assert isinstance(layer['queue'][1], qml.RY)
+        assert isinstance(layer['queue'][2], qml.CNOT)
+        assert isinstance(layer['queue'][3], qml.CNOT)
+        assert isinstance(layer['observable'][0], qml.PauliX)
+        assert isinstance(layer['observable'][1], qml.PauliY)
+        assert isinstance(layer['observable'][2], qml.PauliZ)
+
+        # fourth layer subcircuit
+        layer = res[(5, 6, 7)]
+        assert len(layer['queue']) == 9
+        assert len(layer['observable']) == 3
+        assert isinstance(layer['observable'][0], qml.PauliX)
+        assert isinstance(layer['observable'][1], qml.PauliY)
+        assert isinstance(layer['observable'][2], qml.PauliZ)
+
+    def test_evaluate_subcircuits(self, tol):
+        """Test subcircuits evaluate correctly"""
+        dev = qml.device('default.qubit', wires=2)
+
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        # construct subcircuits
+        circuit = qml.QNode(circuit, dev)
+        circuit.construct_metric_tensor([1, 1, 1])
+
+        a = 0.432
+        b = 0.12
+        c = -0.432
+
+        # evaluate subcircuits
+        circuit.metric_tensor(a, b, c)
+
+        # first parameter subcircuit
+        assert np.allclose(circuit._metric_tensor_subcircuits[(0,)]['result'], 0.25, atol=tol, rtol=0)
```

Comment:
clearer to use the `res`, `expected` pattern here as well

Reply:
done